owlkettle/widgets.nim

```diff
@@ -3616,7 +3616,53 @@ renderable Expander of BaseWidget:
     
     widget.hasLabelWidget = true
     widget.valLabelWidget = child
-    
+
+renderable PasswordEntry of BaseWidget:
+  text: string
+  # menuModel: GtkMenuModel # Not yet supported
+  activatesDefault: bool = true
+  placeholderText: string = "Password"
+  showPeekIcon: bool = true
+  
+  proc activate(password: string) ## Triggered when the user "activated" the entry e.g. by hitting "enter" key while PasswordEntry is in focus. 
+  proc changed(password: string) ## Triggered when the user types in the PasswordEntry.
+  
+  hooks:
+    beforeBuild:
+      state.internalWidget = gtk_password_entry_new()
+    connectEvents:
+      proc activateEventCallback(
+        widget: GtkWidget, 
+        data: ptr EventObj[proc(password: string)]
+      ) {.cdecl.} =
+        let password: string = $gtk_editable_get_text(widget)
+        PasswordEntryState(data[].widget).text = password
+        data[].callback(password)
+        data[].redraw()
+
+      state.connect(state.activate, "activate", activateEventCallback)
+      state.connect(state.changed, "changed", activateEventCallback)
+    disconnectEvents:
+      disconnect(state.internalWidget, state.activate)
+      disconnect(state.internalWidget, state.changed)
+
```

Comment:
hook for `text` seems to be missing. It should be the same as in `Entry`. (Potential refactor: introduce an `Editable` viewable that handles `text` & `changed`).

Reply:
I'll add that refactor to the list of refactors in the discussion to be done in a separate PR